pkg/util/kubernetes/apiserver/util.go

```diff
@@ -61,3 +64,39 @@ func UnstructuredFromWPA(structIn *v1alpha1.WatermarkPodAutoscaler, unstructOut
 	unstructOut.SetUnstructuredContent(content)
 	return nil
 }
+
+func retryHandler(informers map[InformerName]cache.SharedInformer, name InformerName, config retry.Config, maxTotalRetry time.Duration, method func(currentRetry time.Duration, config retry.Config) time.Duration) error {
+	nextTry := config.InitialRetryDelay
+	totalRetry := config.InitialRetryDelay
+	lastTry := false
+	for {
+		ctx, cancel := context.WithTimeout(context.Background(), nextTry)
+		start := time.Now()
+		if !cache.WaitForCacheSync(ctx.Done(), informers[name].HasSynced) {
+			log.Warnf("couldn't sync informer %s in %v", name, time.Now().Sub(start))
+			nextTry = method(nextTry, config)
+			config.RetryCount++
+			log.Warnf("Informers failed to sync, increasing timeout to %s", nextTry)
+			if lastTry {
+				defer cancel()
```

Comment:
you are still defering the cancel. I think the correct choice would be to call the `cancel` at the end of the loop 
```
for:
  if ...
  else ..
  cancel()
```